engine/src/main/java/org/hibernate/search/engine/search/dsl/predicate/impl/MatchIdPredicateContextImpl.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.search.dsl.predicate.impl;
+
+import org.hibernate.search.engine.search.dsl.predicate.MatchIdPredicateContext;
+import org.hibernate.search.engine.search.dsl.predicate.SearchPredicateFactoryContext;
+import org.hibernate.search.engine.search.dsl.predicate.spi.AbstractSearchPredicateTerminalContext;
+import org.hibernate.search.engine.search.predicate.spi.MatchIdPredicateBuilder;
+import org.hibernate.search.engine.search.predicate.spi.SearchPredicateBuilderFactory;
+
+
+class MatchIdPredicateContextImpl<B>
+		extends AbstractSearchPredicateTerminalContext<B>
+		implements MatchIdPredicateContext {
+
+	private final MatchIdPredicateBuilder<B> matchIdBuilder;
+
+	MatchIdPredicateContextImpl(SearchPredicateBuilderFactory<?, B> factory, SearchPredicateFactoryContext factoryContext) {
+		super( factory );
+		this.matchIdBuilder = factory.id();
+	}
+
+	@Override
+	public MatchIdPredicateContext matching(Object value) {
+		matchIdBuilder.value( value );
```

Comment:
I see you allow multiple values, which is nice; wouldn't be useful to also provide a `matchingAny(Collection<String>)` method? From what I can see it can be implemented easily here with a simple loop, without changing the backend.

Reply:
Sure, sounds good